pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	cs "github.com/jetstack/cert-manager/pkg/client/clientset/versioned/typed/certmanager/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	admv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	issuerName        = "selfsigned-issuer"
+	certificateName   = "rook-admission-controller-cert"
+	webhookConfigName = "rook-ceph-webhook"
+)
+
+func fetchorCreateIssuer(ctx context.Context, certMgrClient *cs.CertmanagerV1Client) (*api.Issuer, error) {
+	logger.Infof("Fetching Issuer %s/%s.", namespace, issuerName)
+	issuer, err := certMgrClient.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return createIssuers(ctx, certMgrClient)
+		}
+
+		logger.Errorf("failed to get issuer %s. %v", issuerName, err)
+		return issuer, err
+	}
+
+	return issuer, nil
+}
+
+func createIssuers(ctx context.Context, certMgrClient *cs.CertmanagerV1Client) (*api.Issuer, error) {
+	logger.Infof("Creating Issuer %s/%s.", namespace, issuerName)
+	issuer := &api.Issuer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      issuerName,
+			Namespace: namespace,
+		},
+		Spec: api.IssuerSpec{
+			IssuerConfig: api.IssuerConfig{
+				SelfSigned: &api.SelfSignedIssuer{},
+			},
+		},
+	}
+	issuers, err := certMgrClient.Issuers(namespace).Create(ctx, issuer, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create issuer %s. %v", issuerName, err)
+	}
+
+	sleepTime := 1
+	attempts := 10
+
+	for i := 0; i < attempts; i++ {
+		issuers, err = certMgrClient.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("failed to get issuer %s. %v", issuerName, err)
+			return issuers, err
+		}
+
+		if len(issuers.Status.Conditions) != 0 {
+			if v1.ConditionStatus(issuers.Status.Conditions[0].Reason) == v1.ConditionStatus(api.IssuerConditionReady) {
```

Comment:
The issuer has a different reason. We might need to check for both `reason: isReady` and `status: True`. The status could theoretically be false if the initialization is still in progress.
```
    - lastTransitionTime: "2022-04-08T15:13:09Z"
      observedGeneration: 1
      reason: IsReady
      status: "True"
      type: Ready
```

Reply:
cert-manager `Status.Condition*` values are strange. Checking for both now